lib/netsuite/records/vendor_payment_apply_list.rb

```diff
@@ -0,0 +1,32 @@
+module NetSuite
+  module Records
+    class VendorPaymentApplyList
+      include Support::Fields
+      include Namespaces::TranPurch
+
+      fields :apply
+
+      def initialize(attributes = {})
+        initialize_from_attributes_hash(attributes)
+      end
+
+      def apply=(applies)
+        case applies
+          when Hash
+            self.applies << VendorPaymentApply.new(applies)
+          when Array
+            applies.each { |apply| self.applies << VendorPaymentApply.new(apply) }
```

Comment:
I wrote it this way because this is the pattern used in all of the list
records, e.g. BillingScheduleMilestoneList, CashRefundItemList,
CreditMemoApplyList, InvoiceItemList, etc.  Its a bit strange, but its
consistent :smile:

On Sun, Mar 1, 2015 at 9:11 PM, Jerome Cornet <notifications@github.com>
wrote:

> In lib/netsuite/records/vendor_payment_apply_list.rb
> <https://github.com/prburke/netsuite/pull/4#discussion_r25575511>:
>
> > +    class VendorPaymentApplyList
> > +      include Support::Fields
> > +      include Namespaces::TranPurch
> > +
> > +      fields :apply
> > +
> > +      def initialize(attributes = {})
> > +        initialize_from_attributes_hash(attributes)
> > +      end
> > +
> > +      def apply=(applies)
> > +        case applies
> > +          when Hash
> > +            self.applies << VendorPaymentApply.new(applies)
> > +          when Array
> > +            applies.each { |apply| self.applies << VendorPaymentApply.new(apply) }
>
> self.applies = applies.map { |apply| VendorPaymentApply.new(apply) }
>
> otherwise this will just append to the existing self.applies
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prburke/netsuite/pull/4/files#r25575511>.
>